lib/einhorn/command.rb

```diff
@@ -488,6 +492,40 @@ def self.cull
         Einhorn.log_info("Have too many workers at the current version, so killing off #{excess.length} of them.")
         signal_all("USR2", excess)
       end
+
+      # Cleanup any workers that have been signaled but not shutdown yet.
+      kill_old_upgraded_workers if Einhorn::State.signal_timeout
+    end
+
+    def self.kill_old_upgraded_workers
+      now = Time.now
+      children = Einhorn::State.children.select do |_,c|
+        # Only interested in old, upgraded workers that have been signaled.
```

Comment:
I think the term "upgraded" here might be wrong? Aren't they more like outdated?

Reply:
upgraded meaning a `USR2` was sent to them previously. outdated would be a version difference (I think?)

Reply:
Yeah, to me, "upgraded" means that they're a newer version than the stuff that was running before. If this is for the kicking task, I think in einhorn's view they're still an old version, right?